### PR TITLE
Add fuzzy map search (#43)

### PR DESCRIPTION
**Pull request by ChillerDragon**
Not my proudest code.

I totally broke your functional code and made refresh() go full side effects. Also had to move the promise await out of the inline js. During that refactor I sadly lost the catch with the nice error message.

~~The search is case sensitive and not fuzzy. I will look into that. This horribly refactor should lay the groundwork for easier fuzzy search.~~

:heavy_check_mark: it is now fuzzy

https://user-images.githubusercontent.com/20344300/180169895-b18b95b9-9e98-42f2-aa25-c2394002b088.mp4

### CODE REVIEW

**Comment by k2d222** [comment]
Glad you had a go at this! Didn't wanted to do it myself :eyes: 
The code is not so bad lol

**Comment by ChillerDragon** [comment]
Gimme a min and fuzzy is in

**Comment by ChillerDragon** [comment]
https://user-images.githubusercontent.com/20344300/180174548-4d401c3e-d4bf-4054-8706-998dea126fb0.mp4

File: client/src/ui/routes/lobby.svelte [hunk 1]
```diff
@@ -67,6 +70,12 @@
       }
     }
   }
+
+  // auto focus search box on page load
+  function onload(element) {
+    element.focus()
+  }
```
Comment (k2d222) [inline client/src/ui/routes/lobby.svelte:77]: nice touch

**Pull request merged**